app/src/main/java/uk/gov/mint/MintService.java

```diff
@@ -12,15 +14,28 @@ public MintService(LoadHandler loadHandler) {
         this.loadHandler = loadHandler;
     }
 
+    @Context
+    HttpServletRequest httpServletRequest;
 
     @POST
     @Path("/load")
     public Response load(String payload) {
         try {
-            loadHandler.handle(payload);
+            loadHandler.handle(extractRegisterNameFromRequest(), payload);
             return Response.ok().build();
         } catch (Exception e) {
-            return Response.serverError().entity(e).build();
+            if (e.getCause() instanceof EntryValidator.EntryValidationException) {
+                EntryValidator.EntryValidationException errorCause = (EntryValidator.EntryValidationException) e.getCause();
+                return Response.status(400).entity(errorCause.getMessage() + "\nEntity: '" + errorCause.getEntry().toString() + "'").build();
```

Comment:
Why aren't we using ExceptionMappers?

Also, the extra wrapping and calling `.getCause()` feels like a smell (see related comment about catching & wrapping Throwable).